convert-ga4gh/src/main/java/org/bdgenomics/convert/ga4gh/RecordGroupToReadGroup.java

```diff
@@ -0,0 +1,93 @@
+/**
+ * Licensed to Big Data Genomics (BDG) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The BDG licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.bdgenomics.convert.ga4gh;
+
+import static org.apache.commons.lang3.StringUtils.isNotEmpty;
+
+import javax.annotation.concurrent.Immutable;
+
+import ga4gh.Common.Program;
+
+import ga4gh.Reads.ReadGroup;
+
+import org.bdgenomics.convert.AbstractConverter;
+import org.bdgenomics.convert.ConversionException;
+import org.bdgenomics.convert.ConversionStringency;
+import org.bdgenomics.convert.Converter;
+
+import org.bdgenomics.formats.avro.ProcessingStep;
+import org.bdgenomics.formats.avro.RecordGroup;
+
+import org.slf4j.Logger;
+
+/**
+ * Convert bdg-formats RecordGroup to GA4GH ReadGroup.
+ */
+@Immutable
+final class RecordGroupToReadGroup extends AbstractConverter<RecordGroup, ReadGroup> {
+    /** Convert bdg-formats ProcessingStep to GA4GH Program. */
+    private final Converter<ProcessingStep, Program> processingStepConverter;
+
+    /**
+     * Convert bdg-formats RecordGroup to GA4GH ReadGroup.
+     *
+     * @param processingStepConverter bdg-formats ProcessingStep to GA4GH Program
+     *    converter, must not be null
+     */
+    RecordGroupToReadGroup(final Converter<ProcessingStep, Program> processingStepConverter) {
+        super(RecordGroup.class, ReadGroup.class);
+        checkNotNull(processingStepConverter);
+        this.processingStepConverter = processingStepConverter;
+    }
+
+
+    @Override
+    public ReadGroup convert(final RecordGroup recordGroup,
+                             final ConversionStringency stringency,
+                             final Logger logger) throws ConversionException {
+
+        if (recordGroup == null) {
+            warnOrThrow(recordGroup, "must not be null", null, stringency, logger);
+            return null;
+        }
+
+        ReadGroup.Builder builder = ReadGroup.newBuilder()
+            .setId(isNotEmpty(recordGroup.getName()) ? recordGroup.getName() : "1")
+            .setName(isNotEmpty(recordGroup.getName()) ? recordGroup.getName() : "1");
```

Comment:
Qhat are the IDs and Names the same here?

Reply:
It is noted here

https://github.com/bigdatagenomics/convert/blob/0c964fc9ededdb0d972ada5b4291ca2d64842093/convert-ga4gh/src/main/java/org/bdgenomics/convert/ga4gh/ReadGroupToRecordGroup.java#L71

ga4gh `ReadGroup` has an ID but bdg-formats `RecordGroup` does not.

In ga4gh, the ID is used to join between `ReadAlignment` and `ReadGroup`.  In bdg-formats, the name is used to join between `AlignmentRecord` and `RecordGroup`.  In order to round trip between the two, we need to set `ReadGroup.id` from `RecordGroup.name`.

If `ReadGroup.id` and `ReadGroup.name` are different, on conversion to `RecordGroup` we lose the ID.  If `RecordGroup` had an attributes field, I suppose we could shove the ID in there.  Or alternatively, we refactor bdg-formats `RecordGroup` and `AlignmentRecord` to be closer to ga4gh.

Reply:
Ah, makes sense, thanks!